STL_Extension/include/CGAL/Compact_container.h

```diff
@@ -328,17 +339,8 @@ class Compact_container
     return all_items[block_number].first[index_in_block];
   }
 
-  void swap(Self &c)
-  {
-    std::swap(alloc, c.alloc);
-    std::swap(capacity_, c.capacity_);
-    std::swap(size_, c.size_);
-    std::swap(block_size, c.block_size);
-    std::swap(first_item, c.first_item);
-    std::swap(last_item, c.last_item);
-    std::swap(free_list, c.free_list);
-    all_items.swap(c.all_items);
-    std::swap(time_stamper, c.time_stamper);
+  friend void swap(Compact_container& a, Compact_container b) {
```

Comment:
`noexcept` as well?

Reply:
Yes, it is `noexcept`. I have not tried to add `noexcept`-specifications on all functions, but only on move-constructors and move-assignments. In the end, I no longer use swap for the move-semantic, so there is not need to add `noexcept` on `swap` methods or `swap` functions.

I know the performance benefits of `noexcept` move-constructors and move-assignements. What are the benefits of `noexcept` swap functions?

Reply:
> What are the benefits of `noexcept` swap functions?

They make is_nothrow_swappable_v true? They probably had a reason to create that trait, and I could imagine using it for optimizations similar to noexcept move, in particular when move construction is not noexcept (like std::list with msvc or std::deque with libstdc++), but in practice libstdc++ only uses that trait so it can mark other swap functions noexcept.
It just feels strange that std::swap is noexcept for that type (based on moves) but the swap found by ADL is not.

Reply:
There is a really good reason to create `std::is_nothrow_swappable`, and it is explained in [the "motivation" section of N3619](https://wg21.link/n3619#motivation). But, as you say, it seems to be designed to be used to write `noexcept` specifications for swap functions using other swap functions.

I agree that might be nice to have `noexcept` specifications for more functions in CGAL, but without the immediate performance benefit, I am not willing to do it. Even for adding `noexcept` to `Compact_container::swap`, I have to verify if `std::vector` is nothrow_swappable (it is), and for the concurrent compact container, I do not know if the TBB vector is nothrow swappable. So... I postpone...

Reply:
> I no longer use swap for the move-semantic

You do, the move assignment and move constructor both use swap (the member function).

Reply:
Oh, you are right! For `Compact_container` and `Concurrent_compact_container`, I lazily still rely on `swap`... :cry:

Reply:
> I lazily still rely on `swap`... :cry:

There's nothing wrong with that...